test/test_shared_library.cpp

```diff
@@ -55,3 +55,14 @@ TEST(test_shared_library, failed_test) {
     FAIL();
   }
 }
+
+TEST(test_get_platform_library_name, failed_test) {
+  // create a string bigger than the internal buffer
+  int length = 2000;
+  std::string str;
+  for (int i = 0; i < length; i++) {
+    str.push_back('A');
+  }
+  std::string library_path = rcpputils::get_platform_library_name(str);
+  EXPECT_TRUE(library_path.empty());
```

Comment:
This behavior (returning an empty string) doesn't seem to be documented, or am I missing the point of this test? Is it because the input str doesn't exist locally (it seems like that shouldn't matter)? I don't understand, even after looking at the rcutils implementation.

Reply:
The idea it's to use a library_name bigger than 1024 characters ( the size of the internal buffer ), then `rcutils_get_platform_library_name` should return a `RCUTILS_RET_ERROR`

I will document about the maximum size and empty string

Reply:
But if there's an rcutils error code, will this not throw... I don't see any case where the string is empty.

Reply:
That's true :man_facepalming:

Reply:
This may be related to how you're using `rcutils_snprintf`? The input being too long is not an error case, it is supposed to truncate the string instead, see:

https://github.com/ros2/rcutils/blob/04456c7540362cf60a01b19d0e505fded4156795/test/test_snprintf.cpp#L32-L34

Reply:
As far as I see, there is not tests in which the buffer is smaller than the string to be copied. which is the case that I'm trying to cover